vcd/resource_vcd_vapp_vm.go

```diff
@@ -135,6 +177,34 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 			Computed:    true,
 			Description: "The number of cores per socket",
 		},
+		"cpu_reservation": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "The amount of Mhz reservation on the underlying virtualization infrastructure",
+			ValidateFunc: validateMultipleOf4(),
+		},
+		"cpu_priority_type": &schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Pre-determined relative priorities according to which the non-reserved portion of this resource is made available to the virtualized workload",
+			ValidateFunc: validation.StringInSlice([]string{"LOW", "NORMAL", "HIGH", "CUSTOM"}, false),
+		},
+		"cpu_shares": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Custom priority for the resource. This is a read-only, unless the share level is CUSTOM",
+			ValidateFunc: validateMultipleOf4(),
+		},
+		"cpu_limit": &schema.Schema{
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "The limit for how much of CPU can be consumed on the underlying virtualization infrastructure. This is only valid when the resource allocation is not unlimited.",
+			ValidateFunc: validateMultipleOf4(),
```

Comment:
Sounds like it will also not support `-1`

Reply:
removed as API allows any value, not only `of 4`. Also allows -1